cpp/Platform.Setters/SetterBase.h

```diff
@@ -1,16 +1,16 @@
-﻿namespace Platform::Setters
-{
-    template <typename ...> class SetterBase;
-    template <typename TResult> class SetterBase<TResult> : public ISetter<TResult>
-    {
-        protected: TResult _result = 0;
-        
-        public: TResult Result() { return _result; }
-        
-        protected: SetterBase() { }
-    
-        protected: SetterBase(TResult defaultValue) { _result = defaultValue; }
-        
-        public: void Set(TResult value) { _result = value; }
-    };
-}
+﻿namespace Platform::Setters
+{
+    template <typename ...> class SetterBase;
+    template <typename TResult> class SetterBase<TResult>
+    {
+        protected: TResult _result {};

```

Comment:
@Konard maybe add 
```cpp
static_assert(std::default_initializable<TResult>)
```
Or use `std::optional`

Reply:
No `std::optional` please. Please no. Let the user decide what TResult is.

Reply:
@Mitron57 add static assert to the constructor with least number of arguments, please.

Reply:
We also should check for compliance with ISetter concept.